### DISCUSSION TITLE
Tell if an object changed and should be saved

### DISCUSSION THREAD

**Discussion by pawamoy** [collerek/ormar#276] [Q&A]
I see in the docs that there is a `saved` property: https://collerek.github.io/ormar/api/models/new-basemodel/#saved
But I'm not sure to understand if `saved` will return False if I do something like `obj.field = obj.field` (actually *not* changing anything, but still setting an attribute). Or if Python is clever enough to make it a no-op, something like `a = obj.field; obj.field = a`.
The goal is to `update` only if the object has changed, to avoid extra queries against the database. But maybe update already checks that the object has changed?

**Comment by collerek** [comment]
As of now, it's pretty naive as I didn't want to issue db queries.
So it only tracks the changes applied on the object -> regardless of the values, so even if you re-assign the same value. 

A more detailed explanation is here: https://collerek.github.io/ormar/models/#model-save-status

So it will fire an update after you reassign the field. The database backends usually have change comparison implemented so the database does not actually update the column if the value is the same (depending on the backend), but the `ormar` query will fire.

That is a potential enhancement since field access is already happening through descriptors, that can check the previous value, some additional work would be required for `__init__`.

**Reply by pawamoy** [reply]
I see, that's what I suspected, thank you for the explanation :slightly_smiling_face: 
In the meantime I can easily write a condition myself to check if the 2-3 fields I'm assigning have changed or not, so no urgency here!
